DirectProgramming/DPC++/StructuredGrids/particle-diffusion/src/motionsim_kernel.cpp

```diff
@@ -0,0 +1,236 @@
+//==============================================================
+// Copyright © 2019 Intel Corporation
+//
+// SPDX-License-Identifier: MIT
+// =============================================================
+
+// motionsim_kernel:
+// Intel® oneAPI DPC++ Language Basics Using a Monte Carlo
+// Simulation (Device)
+//
+
+#include "motionsim.hpp"
+namespace oneapi {}
+using namespace oneapi;
```

Comment:
@pmpeter1 - Is there a specific rule against "using oneapi" in the cpp code?

Reply:
The sample style guidelines say it's OK (and often preferred) to use "using" in the .cpp files explicitly.   It should not be in a header file.     This is to make it obvious you are introducing this short-cut, and to keep the code simpler.

I'm not exactly sure why we are introducing an empty namespace and then opening it here.   So the question is why like 13?

Reply:
This looks like a temporary measure to support beta08 oneMKL (where APIs are in the mkl namespace) and beta09 oneMKL (where APIs are in oneapi::mkl) simultaneously.

Reply:
Honestly, I have no idea as to the implications of the syntax configuration of lines 13 and 14, I was just following what Alberto Villarreal did in his version to declare the namespace

Reply:
I will be happy to change it if there is a proposed, preferred way of declaration.